ambari-server/src/main/java/org/apache/ambari/server/ServiceComponentNotFoundException.java

```diff
@@ -32,4 +32,14 @@ public ServiceComponentNotFoundException (String clusterName,
         + ", serviceComponentName=" + serviceComponentName);
   }
 
+  public ServiceComponentNotFoundException (String clusterName,
+      String serviceName, String serviceType, String serviceGroupName, Long serviceComponentId) {
+    super("ServiceComponent not found"
+            + ", clusterName=" + clusterName
+            + ", serviceGroupName=" + serviceGroupName
+            + ", serviceName=" + serviceName
+            + ", serviceType=" + serviceType
+            + ", serviceComponentId=" + serviceComponentId);
+  }
+
 }
```

Comment:
Where is serviceComponentId stored in the DB?

Reply:
It should come from *servicecomponentdesiredstate* table.